src/System Application/App/AI/src/Azure OpenAI/AzureOpenAIImpl.Codeunit.al

```diff
@@ -388,38 +389,43 @@ codeunit 7772 "Azure OpenAI Impl"
         FunctionResult: Variant;
     begin
         if Functions.Count = 0 then
-            exit;
+            exit(false);
 
         Functions.Get(0, Token);
         Function := Token.AsObject();
 
         if Function.Get('type', Token) then begin
             if Token.AsValue().AsText() <> 'function' then
-                exit;
+                exit(false);
         end else
-            exit;
+            exit(false);
 
         if Function.Get('function', Token) then
             Function := Token.AsObject()
         else
-            exit;
+            exit(false);
 
         if Function.Get('name', Token) then
             FunctionName := Token.AsValue().AsText()
         else
-            exit;
+            exit(false);
 
         if Function.Get('arguments', Token) then
             // Arguments are stored as a string in the JSON
             Arguments.ReadFrom(Token.AsValue().AsText());
 
         if ChatMessages.GetFunctionTool(FunctionName, AOAIFunction) then
```

Comment:
Isn't this same as 
        if ChatMessages.GetFunctionTool(FunctionName, AOAIFunction) then begin
            if TryExecuteFunction(AOAIFunction, Arguments, FunctionResult) then
                AOAIFunctionResponse.SetFunctionCallingResponse(true, true, AOAIFunction.GetName(), FunctionResult, '', '')
            else
                AOAIFunctionResponse.SetFunctionCallingResponse(true, false, AOAIFunction.GetName(), FunctionResult, GetLastErrorText(), GetLastErrorCallStack());
        end else
            AOAIFunctionResponse.SetFunctionCallingResponse(true, false, FunctionName, FunctionResult, StrSubstNo(FunctionCallingFunctionNotFoundErr, FunctionName), '');

        exit(true);

Reply:
Yes, but feels bit wrong to use a catch all to return `true`...